includes/wikia/services/ImagesService.class.php

```diff
@@ -157,28 +188,30 @@ private static function isDataTagImage($url) {
 	 */
 	public static function getThumbUrlFromFileUrl($imageUrl, $destSize, $newExtension = null) {
 		if (!empty($imageUrl)) {
-			if ( !self::IsExternalThumbnailUrl($imageUrl) ) {
-				$imageUrl = str_replace('/images/', '/images/thumb/', $imageUrl);
+			if ( VignetteRequest::isVignetteUrl( $imageUrl ) ) {
+				list( $width, $height ) = self::parseThumbDestSize( $destSize );
+				$imageUrl = self::vignetteOriginalToThumb( $imageUrl, $width, $height );
 			} else {
-				$imageUrl = $imageUrl;
-			}
-
+				if ( !self::IsExternalThumbnailUrl( $imageUrl ) ) {
+					$imageUrl = str_replace( '/images/', '/images/thumb/', $imageUrl );
+				}
 
-			/**
-			 * url is virtual base for thumbnail, so
-			 *
-			 * - get last part of path
-			 * - add it as thumbnail file prefixed with widthpx
-			 */
-			$parts = explode( "/", $imageUrl );
-			$file = array_pop( $parts );
+				/**
+				 * url is virtual base for thumbnail, so
+				 *
+				 * - get last part of path
+				 * - add it as thumbnail file prefixed with widthpx
+				 */
+				$parts = explode( "/", $imageUrl );
+				$file = array_pop( $parts );
+
+				if ( ctype_digit( (string)$destSize ) ) {
+					$destSize .= 'px';
+				}
```

Comment:
Indentation is now one level deeper, since we're checking if the url is a vignette url.